### PR TITLE
SSCS-7079 Interloc – Clear the interlocReviewState flag if case goes …

### PR DESCRIPTION
**Pull request by srochani**
…dormant or void

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -13,7 +13,7 @@ plugins {
     id 'pmd'
     id 'org.sonarqube' version '2.8'
     id 'jacoco'
-    id 'org.springframework.boot' version '2.2.0.RELEASE'
+    id 'org.springframework.boot' version '2.2.3.RELEASE'
```
Comment (iguannago) [inline build.gradle]: you can safely move onto the 2.2.4 here.

File: build.gradle [hunk 2]
```diff
@@ -258,15 +258,15 @@ dependencies {
...
+    compile group: 'uk.gov.hmcts.reform', name: 'sscs-common', version: '3.1.17-RC-SSCS-7079'
     compile group: 'uk.gov.hmcts.reform', name: 'sscs-pdf-email-common', version: '1.2.4'
 
     testCompile group: 'io.rest-assured', name: 'rest-assured'
 
     compile 'org.projectlombok:lombok:1.18.10'
     annotationProcessor 'org.projectlombok:lombok:1.18.10'
 
-    testCompile group: 'org.springframework.boot', name: 'spring-boot-starter-test', version: '2.2.0.RELEASE'
+    testCompile group: 'org.springframework.boot', name: 'spring-boot-starter-test', version: '2.2.3.RELEASE'
```
Comment (iguannago) [inline build.gradle]: move onto the 2.2.4 too, please.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminappealwithdrawn/AdminAppealWithdrawnAboutToSubmitHandler.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+
+@Service
+@Slf4j
+public class AdminAppealWithdrawnAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminappealwithdrawn/AdminAppealWithdrawnAboutToSubmitHandler.java]: For the future, when you have to set a field you don't need to create new handlers. You can reuse this handler: EventToFieldPreSubmitCallbackHandler.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminappealwithdrawn/AdminAppealWithdrawnAboutToSubmitHandler.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.ABOUT_TO_SUBMIT)
+                && callback.getEvent() == EventType.ADMIN_APPEAL_WITHDRAWN;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminappealwithdrawn/AdminAppealWithdrawnAboutToSubmitHandler.java]: For the future, when you have to set a field you don't need to create new handlers. You can reuse this handler: EventToFieldPreSubmitCallbackHandler.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminsendtodormantappealstate/AdminSendToDormantAppealStateAboutToSubmitHandler.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+@Service
+@Slf4j
+public class AdminSendToDormantAppealStateAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminsendtodormantappealstate/AdminSendToDormantAppealStateAboutToSubmitHandler.java]: For the future, when you have to set a field you don't need to create new handlers. You can reuse this handler: EventToFieldPreSubmitCallbackHandler.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/appealdormant/AppealDormantAboutToSubmitHandler.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+
+@Service
+@Slf4j
+public class AppealDormantAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/appealdormant/AppealDormantAboutToSubmitHandler.java]: For the future, when you have to set a field you don't need to create new handlers. You can reuse this handler: EventToFieldPreSubmitCallbackHandler.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/appeallapsed/AppealLapsedAboutToSubmitHandler.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+
+@Service
+@Slf4j
+public class AppealLapsedAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/appeallapsed/AppealLapsedAboutToSubmitHandler.java]: For the future, when you have to set a field you don't need to create new handlers. You can reuse this handler: EventToFieldPreSubmitCallbackHandler.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/appealwithdrawn/AppealWithdrawnAboutToSubmitHandler.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+
+@Service
+@Slf4j
+public class AppealWithdrawnAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/appealwithdrawn/AppealWithdrawnAboutToSubmitHandler.java]: For the future, when you have to set a field you don't need to create new handlers. You can reuse this handler: EventToFieldPreSubmitCallbackHandler.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/confirmlapsed/ConfirmLapsedAboutToSubmitHandler.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+@Service
+@Slf4j
+public class ConfirmLapsedAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/confirmlapsed/ConfirmLapsedAboutToSubmitHandler.java]: For the future, when you have to set a field you don't need to create new handlers. You can reuse this handler: EventToFieldPreSubmitCallbackHandler.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/hmctslapsecase/HmctsLapseCaseAboutToSubmitHandler.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+
+@Service
+@Slf4j
+public class HmctsLapseCaseAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/hmctslapsecase/HmctsLapseCaseAboutToSubmitHandler.java]: For the future, when you have to set a field you don't need to create new handlers. You can reuse this handler: EventToFieldPreSubmitCallbackHandler.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminappealwithdrawn/AdminAppealWithdrawnAboutToSubmitHandler.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+
+        log.info("Setting InterlocReviewState to N/A for case id {} "
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminappealwithdrawn/AdminAppealWithdrawnAboutToSubmitHandler.java]: The log message doesn't match what it is actually doing? Can we update so this says 'Setting InterlocReviewState to null for case id {} ' - This is in every handler

Comment (srochani) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminappealwithdrawn/AdminAppealWithdrawnAboutToSubmitHandler.java]: Updated in every handler

**Pull request merged**